### PR TITLE
Collage video block focus style update

### PR DESCRIPTION
**Pull request by ludoboludo**
**Why are these changes introduced?**

The focus style on the video block is updated to use the same kinda logic (focus/focus-visible) as the cards.

**What approach did you take?**

Removed `focus-within` and used the `focus, focus-visible` approach.

**Other considerations**

**Demo links**

- [Store](https://os2-demo.myshopify.com/?preview_theme_id=127083380758)
- [Editor](https://os2-demo.myshopify.com/admin/themes/127083380758/editor)

**Checklist**
- [ ] Followed [theme code principles](https://github.com/Shopify/dawn/blob/main/.github/CONTRIBUTING.md#theme-code-principles)
- [ ] Linted with [Theme Check](https://github.com/Shopify/theme-check)
- [ ] Tested on [mobile](https://shopify.dev/themes/store/requirements#mobile-browser-requirements)
- [ ] Tested on [multiple browsers](https://shopify.dev/themes/store/requirements#desktop-browser-requirements)
- [ ] Tested for [accessibility](https://shopify.dev/themes/best-practices/accessibility)

### CODE REVIEW

**Review by tyleralsbury — APPROVED** [review]
Looks good - Tested and the focus style only appears when focusing manually or when the modal sets the focus to the video element on close.  It does not appear when clicked, which is the goal 👍

File: assets/collage.css
Comment (martinamarien) [inline assets/collage.css:159]: In safari, the focus looks different for the selected video vs. the onboarding state. If you click outside of the safari window, the grey border is displayed, but not when tabbing. The focus state looks right for the regular video section.

https://user-images.githubusercontent.com/26778637/148977678-67f95c60-6cff-42e1-97aa-ab4c066adadc.mov


Screen Shot 2022-01-11 at 11 06 00 AM

Screen Shot 2022-01-11 at 11 06 16 AM

**Review by martinamarien — CHANGES_REQUESTED** [review]
The border in Safari doesn't seem right to me. It doesn't seem to have the more prominent/contrasted focus state.

File: assets/collage.css
Comment (ludoboludo) [reply assets/collage.css:164]: Ah I know why. I grouped selectors for `:focus` and `:focus-within` which I should not because `:focus-within` isn't supported in safari so the styling for `:focus` is not applied. I separated them and it should now be good 🙂

Comment (melissaperreault) [reply assets/collage.css:164]: Related to Safari, I noticed the focus is not following the UI shape here, I can open an issue to not busy up this PR https://screenshot.click/22-01-mtov6-l1wzh.png

Comment (tyleralsbury) [reply assets/collage.css:164]: It may be that Safari is being more strict about the actual CSS spec with regards to outlines (which are not really for decorative use, technically).

**Review by melissaperreault — COMMENTED** [review]
Thanks Ludo! Noticed this detail where the outline is not applied the same way as on other cards ([Video](https://screenshot.click/22-01-fbzuu-u9eh3.mp4))

**Review by tyleralsbury — DISMISSED** [review]
Tested on desktop and mobile, with different settings for borders and shadows for cards. Different colour schemes as well.  Everything looks good to me.  👍

File: assets/collage.css
Comment (martinamarien) [reply assets/collage.css:174]: > From what I can tell most of the focus outlines/shadows, I think all of them actually, aren't receiving the border radius like in other browsers. It's specific to Safari.

Yeah. I talked to Ken and we need to convert them to box-shadow because `outline` doesn't inherit the border-radius.

**Comment by martinamarien** [comment]
@melissaperreault My Product media PR will fix the focus state for video. You can see it fixed on this [homepage](https://os2-demo.myshopify.com/admin/themes/127066177558/editor). 

One thing to note is that we will need to add an override for border radius on focus. But we can fix that easily. I've also updated model focus to behave the same way.

File: assets/collage.css
Comment (martinamarien) [inline assets/collage.css:174]: I don't think you don't need this class here.
This class is only used if there is no-js and we have styling in our global css file that should handle this: 

```
.no-js :focus:not(:focus-visible) {
    outline: 0;
    box-shadow: none;
}
```

File: assets/collage.css
Comment (martinamarien) [reply assets/collage.css:160]: I would update for video, and if there is a separate selector for card then, there as well if it's not too much effort. It's what we are doing for the pills selectors already. So it would be good to continue to do that with the outline approach. Even though it doesn't benefit safari users, the overall UX/UI feels better when the outline follows the border-radius of the card.

**Review by martinamarien — DISMISSED** [review]
Other than my one comment about box-shadow. Looks good. I'll defer to Meli for the decision on whether to remove it or not. If she gives the 👍🏻  then we can ship with a another dev approval. If not, ping me for another review 🙂

**Review by tyleralsbury — DISMISSED** [review]
Approved pending Meli's feedback on the most recent comment.

**Review by martinamarien — APPROVED** [review]

**Review by tyleralsbury — APPROVED** [review]

**Review by melissaperreault — APPROVED** [review]
Thanks Ludo! 🎉 
We can ship 🚢 🚢 

One tiny detail I noticed that we can tackle later is about the change in focus color depending on setting combination ([Video](https://screenshot.click/22-01-v5e2a-u2evb.mp4))

**Review by tyleralsbury — APPROVED** [review]
Tested a few styles and it all feels good to me.

I noticed that the cards don't have gradients when a gradient colour scheme selected.  Is this intended or should we also incorporate the gradient colour schemes into the cards?  Not for this PR, but to maybe spin off another issue for later.

cc @melissaperreault

**Review by martinamarien — APPROVED** [review]
Looks good. 
Let's create an issue to look at focus states for video in the collage, if we haven't already

**Comment by ludoboludo** [comment]
Issue created: https://github.com/Shopify/dawn/issues/1202

**Pull request merged**
